core/config.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+import json
+from pathlib import Path
+from typing import List, Dict, Any, Union
+
+import attr
+
+
+@attr.s(auto_attribs=True)
+class AnnouncementChannels:
+    youtube: List[str]
+    # TODO: Depending on Twitch channel structure, this could be a class
+    twitch: Dict[str, Any]
+    publish: List[str]
+
+    def get_json(self) -> Dict[str, Any]:
+        return attr.asdict(self)
+
+    @classmethod
+    def factory(
+            cls,
+            youtube_channels: List[str],
+            twitch_channels: Dict[str, Any],
+            publish_channels: List[str],
+    ) -> AnnouncementChannels:
+        announcement_channels = cls(
+            youtube=youtube_channels,
+            twitch=twitch_channels,
+            publish=publish_channels,
+        )
+
+        return announcement_channels
+
+
+@attr.s(auto_attribs=True)
+class BotConfig:
+    announcement_channels: AnnouncementChannels
+    vod_count: int
+    youtube_text_channel_id: int
+
+    @classmethod
+    def factory(
+            cls,
+            vod_count: int,
+            youtube_text_channel_id: int,
+            youtube_channels: List[str] = None,
+            twitch_channels: Dict[str, Any] = None,
+            publish_channels: List[str] = None,
+    ) -> BotConfig:
+        bot_config = cls(
+            announcement_channels=AnnouncementChannels.factory(
+                youtube_channels=youtube_channels if youtube_channels else [],
+                twitch_channels=twitch_channels if twitch_channels else dict(),
+                publish_channels=publish_channels if publish_channels else [],
+            ),
+            vod_count=vod_count,
+            youtube_text_channel_id=youtube_text_channel_id,
+        )
+
+        return bot_config
+
+    def get_json(self) -> Dict[str, Any]:
+        return attr.asdict(self)
+
+    def get_store_json(self) -> Dict[str, Any]:
+        return {
+            "youtube_announcement_channels": self.announcement_channels.youtube,
+            "vod_count": self.vod_count,
+            "yt_text_channel_id": self.youtube_text_channel_id,
+            "twitch_announcement_channels": self.announcement_channels.twitch,
+            "publish_announcement_channels": self.announcement_channels.publish
+        }
```

Comment:
I'd also be in favor of changing the name to `config.json` rather than `config_data.json` but that might be out of scope too.

Reply:
Renaming is fine, we can make that change on this PR. Regarding the format, either way works. I personally prefer the second one, as sorting it by feature seems more intuitive but I don't feel super strongly, up to y'all

Reply:
I see that the structure was changed here, thanks! Though I'm not sure why we're still returning a dictionary in the old format here? If it's for the legacy configuration retrieval, I think per https://github.com/Splatoon-Stronghold/cas-stronghold-bot/pull/14#discussion_r1716234131 it should be changed to get values directly from BotConfig rather than via a dictionary.